src/app.js

```diff
@@ -1 +1,50 @@
 'use strict';
+
+const { askNumber, writeResult, finishGame } = require('./terminal');
+const { countBullsAndCows } = require('./bullsAndCows');
+const { NUMBER_OF_DIGITS } = require('./constants');
+
+const isUniqueDigits = (numberArray) => {
+  const set = new Set(numberArray);
+
+  return numberArray.length === set.size;
+};
+
+const checkTurn = (userNumber) => {
+  if (isNaN(userNumber)) {
+    writeResult('You entered invalid number');
+    finishGame();
+
+    return;
+  }
+
+  if (userNumber.length !== NUMBER_OF_DIGITS) {
+    writeResult(`Number should contains ${NUMBER_OF_DIGITS} digits`);
```

Comment:
Move these string to constants too: `Number should contains ${NUMBER_OF_DIGITS} digits`)